models/api-model.js

```diff
@@ -0,0 +1,13 @@
+const db = require("../db/connection");
+const { fs, promises } = require("fs");
+
+exports.selectApis = () => {
+  return promises
+    .readFile("./endpoints.json", "utf-8")
+    .then((endpoints) => {
+      return JSON.parse(endpoints);
+    })
+    .catch((err) => {
+      next(err);
+    });
```

Comment:
Model functions are specifically for interacting with the database. You don't need a model for this, not do model functions need catch blocks if your controller has one.
The endpoints file is already in a format that the client can receive so here you've read a json, to a string, parsed it back to an object to a controller to send it back as a json. A few steps can be skipped here...